service/Abstractions/ContentStorage/IContentStorage.cs

```diff
@@ -90,4 +92,20 @@ public interface IContentStorage
         string fileName,
         bool logErrIfNotFound = true,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Fetch file info from storage
+    /// </summary>
+    /// <param name="index">Index name</param>
+    /// <param name="documentId">Document ID</param>
+    /// <param name="fileName"></param>
+    /// <param name="logErrIfNotFound">Whether to log an error if the file does not exist. An exception will be raised anyway.</param>
+    /// <param name="cancellationToken">Async task cancellation token</param>
+    /// <returns>File content</returns>
+    Task<IContentFile> FileInfoAsync(
```

Comment:
1. rename to `GetFileInfoAsync`
2. why do we need this new method?
3. why return IContentFile (an interface) rather than a specific type?